middleware_oauth.go

```diff
@@ -3,17 +3,18 @@ package main
 import (
 	"encoding/json"
 
+	"errors"
+	"net/http"
```

Comment:
Good that you grouped your imports in blocks like `stdlib`, `external dependencies`, `project packages` separated with empty line 👍 
But it looks like you forgot to remove empty line on line 5